x/stakeibc/keeper/msg_server_change_validator_weight.go

```diff
@@ -15,28 +13,17 @@ func (k msgServer) ChangeValidatorWeight(goCtx context.Context, msg *types.MsgCh
 
 	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
 	if !found {
-		k.Logger(ctx).Error(fmt.Sprintf("Host Zone %s not found", msg.HostZone))
 		return nil, types.ErrInvalidHostZone
 	}
 
 	validators := hostZone.Validators
 	for _, validator := range validators {
-		if validator.GetAddress() == msg.ValAddr {
-
-			// when changing a weight from 0 to non-zero, make sure we have space in the val set for this new validator
-			if validator.Weight == 0 && msg.Weight > 0 {
-				err := k.ConfirmValSetHasSpace(ctx, validators)
-				if err != nil {
-					return nil, errorsmod.Wrap(types.ErrMaxNumValidators, "cannot set val weight from zero to nonzero on host zone")
-				}
-			}
+		if validator.Address == msg.ValAddr {
```

Comment:
Do we really want to allow every validator?  Even if we now will have most of them for the possible LSM token locations do we want the ability to have a blacklist of validators we just don't trust because they have been slashed so many times in the past?  Also it isn't possible for fake validators to exist in this pool right so no need to do some sort of check here that they are real?

Reply:
Validators will only be added and removed through governance so there's no need to blacklist, we can just make sure to not include them in the proposal!